numbat/src/prefix_transformer.rs

```diff
@@ -280,13 +280,15 @@ impl Transformer {
         })
     }
 
-    pub fn transform(
+    pub fn transform<'a>(
         &mut self,
-        statements: impl IntoIterator<Item = Statement>,
-    ) -> Result<Vec<Statement>> {
-        statements
-            .into_iter()
-            .map(|statement| self.transform_statement(statement))
-            .collect()
+        statements: impl IntoIterator<Item = Statement<'a>>,
+    ) -> Result<Vec<Statement<'a>>> {
+        let statements = statements.into_iter();
+        let mut ans = Vec::with_capacity(statements.size_hint().1.unwrap_or(0));
+        for s in statements {
+            ans.push(self.transform_statement(s)?);
+        }
```

Comment:
Is this a performance improvement or was it somehow necessary to rewrite this iterator expression into a loop for another reason?

Reply:
Probably a vestige of me trying to fix a borrow checker error. Reverted.